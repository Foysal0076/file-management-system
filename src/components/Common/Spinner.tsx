type Props = {
  className?: string
}

const Spinner = ({ className = 'text-neutral-500' }: Props) => {
  return (
    <div
      className={`${className} border-current border-r-transparent inline-block h-8 w-8 animate-spin rounded-full border-4 border-solid align-[-0.125em] text-neutral-400 motion-reduce:animate-[spin_1.5s_linear_infinite]`}
      role='status'>
      <span className='!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]'>
        Loading...
      </span>
    </div>
  )
}

export default Spinner
