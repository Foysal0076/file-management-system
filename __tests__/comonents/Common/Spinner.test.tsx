import { render, screen } from '@testing-library/react'

import Spinner from '@/components/Common/Spinner'

describe('Spinner', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Spinner className='h-4 w-4 text-info-800' />)
    expect(container).toMatchSnapshot()
  })

  it('renders with the default className', () => {
    render(<Spinner />)
    const spinner = screen.getByRole('status')
    expect(spinner).toHaveClass('text-neutral-500')
    expect(spinner).toHaveTextContent('Loading...')
  })

  it('renders with a custom className', () => {
    render(<Spinner className='custom-spinner' />)
    const spinner = screen.getByRole('status')
    expect(spinner).toHaveClass('custom-spinner')
    expect(spinner).toHaveTextContent('Loading...')
  })
})
