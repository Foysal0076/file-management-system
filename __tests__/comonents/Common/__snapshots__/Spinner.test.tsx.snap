// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner matches the snapshot 1`] = `
<div>
  <div
    class="h-4 w-4 text-info-800 border-current border-r-transparent inline-block h-8 w-8 animate-spin rounded-full border-4 border-solid align-[-0.125em] text-neutral-400 motion-reduce:animate-[spin_1.5s_linear_infinite]"
    role="status"
  >
    <span
      class="!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]"
    >
      Loading...
    </span>
  </div>
</div>
`;
